src/transformation/utils/lua-ast.ts

```diff
@@ -129,6 +128,17 @@ export function createHoistableVariableDeclarationStatement(
     return declaration;
 }
 
+function hasMultipleReferences(scope: Scope, identifiers: lua.Identifier | lua.Identifier[]) {
+    const scopeSymbols = scope.referencedSymbols;
+    if (!scopeSymbols) {
+        return false;
+    }
+
+    const referenceLists = castArray(identifiers).map(i => i.symbolId && scopeSymbols.get(i.symbolId));
+
+    return referenceLists.some(symbolRefs => symbolRefs && symbolRefs.length > 1);
```

Comment:
Maybe
```suggestion
    return referenceLists.some(symbolRefs => symbolRefs?.length > 1);
```

Reply:
This change won't work as `symbolRefs?.length` resolves to `number | undefined` which can't be compared to a number like that. I could change it to something like `(symbolRefs?.length || 0) > 1`, but I feel like it looses clarity that way.